core/src/toga/app.py

```diff
@@ -520,11 +520,60 @@ def current_window(self):
         """Return the currently active content window."""
         return self._impl.current_window().interface
 
+    @property
+    def maximized(self):
+        """Returns True if any of the app's windows are maximized."""
+        return any(window.maximized for window in self.windows)
+
+    @maximized.setter
+    def maximized(self, maximize: bool):
+        if maximize:
+            for window in self.windows:
+                window.maximized = True
+        else:
+            for window in self.windows:
+                window.maximized = False
+
+    @property
+    def minimized(self):
+        """Returns True only if all of the app's windows are minimized."""
+        return all(window.minimized for window in self.windows)
+
+    @minimized.setter
+    def minimized(self, minimize: bool):
+        if minimize:
+            for window in self.windows:
+                window.minimized = True
+        else:
+            for window in self.windows:
+                window.minimized = False
+
+    @property
+    def full_screen(self):
+        """Return True if any of the app's window are in full screen."""
+        return any(window.full_screen for window in self.windows)
+
+    @full_screen.setter
+    def full_screen(self, full_screen):
+        if full_screen:
+            for window in self.windows:
+                window.full_screen = True
+        else:
+            for window in self.windows:
+                window.full_screen = False
```

Comment:
This approach won't work with the Cocoa implementation. This needs to be done in a batch, so that Cocoa can assign windows to screens.

This is likely true of other platforms as well - there's no point making all the windows full screen on the same screen.

Reply:
I am not sure that I understand you correctly. 

Doing`toga.App.full_screen = True` means that all the windows of the app will be set to full screen. If only selected windows need to be full screened then the user can do `window.full_screen = True` on those windows.

Also what do you mean by "assign window to screens"?  Are you taking about multi-monitor setup or something similar?

Reply:
I'm saying look at the existing implementation of the exiting feature on Cocoa. The API that you've proposed here won't work.

And yes - it's the multi-monitor situation that is the problem here. The full-screen feature exists *specifically* because of multi-screen support. It was added to support [Podium](https://github.com/beeware/podium) - where you have 2 screens - a slide presentation screen, and a speakers note screen. But - you can have multiple "presentation" documents open, and you only want 2 windows to be made full screen, and you need a specific window to go to screen 1.